Dynamics/euler.py

```diff
@@ -91,9 +94,9 @@ def update_position_cartesian(assembly, cartesianDerivatives, angularDerivatives
     """
 
     dt = time_step
+    
+    if iter_num == 0 or options.dynamics.propagator == 'legacy_euler':
```

Comment:
Updated the develop branch previously to use options.current_iter to avoid passing in iteration numbers - can streamline when fixing conflicts when merging.

Reply:
Here iter_num represents iterations since fragmentation as for other integration methods there are issues applying multi-step integration across the CoG discontinuity caused by fragmentation and this keeps legacy_bwd in line with bwd and the AB(N) methods. It's something of a tradeoff whether to be 1st order for fragmentation or have weird behaviour, I'm honestly not sure which is preferrable.

Reply:
So it is just the case of initializing the dynamics of new fragments (in their first instance as individual assemblies) with first order dynamics if we are running legacy approach to dynamics? I think it make sense if it is causing issues for the fragment to inherent the parent/its last iteration as part of parent assembly state. I see the input to the function as post_event_iter so it makes sense - we could probably update the naming so it is explicitly referencing fragmentation, but it doesn't really matter that much if its well defined in the class function.

Reply:
How about?

```suggestion
 # Need to be 1st order at the simulation start and immediately after fragmentation
    if post_event_iter == 0 or options.dynamics.propagator == 'legacy_euler':
```
(and obviously rename in the function argument as well)

Reply:
Yeah I think that works well!